test_framework/terraform/harvester/ubuntu/main.tf

```diff
@@ -0,0 +1,146 @@
+terraform {
+  required_providers {
+    rancher2 = {
+      source  = "rancher/rancher2"
+      version = "~> 4.1.0"
+    }
+  }
+}
+
+provider "rancher2" {
+  api_url   = var.lab_url
+  insecure  = true
+  access_key = var.lab_access_key
+  secret_key = var.lab_secret_key
+}
+
+resource "random_string" "random_suffix" {
+  length           = 8
+  special          = false
+  lower            = true
+  upper            = false
+}
+
+data "rancher2_cluster_v2" "hal-cluster" {
+  name = "hal"
+}
+
+resource "rancher2_cloud_credential" "e2e-credential" {
+  name = "e2e-credential-${random_string.random_suffix.id}"
+  harvester_credential_config {
+    cluster_id = data.rancher2_cluster_v2.hal-cluster.cluster_v1_id
+    cluster_type = "imported"
+    kubeconfig_content = data.rancher2_cluster_v2.hal-cluster.kube_config
+  }
+}
+
+resource "rancher2_machine_config_v2" "e2e-machine-config" {
+
+  generate_name = "e2e-machine-config-${random_string.random_suffix.id}"
+
+  harvester_config {
+
+    vm_namespace = "longhorn-qa"
+
+    cpu_count = "4"
+    memory_size = "8"
+
+    disk_info = <<EOF
+    {
+        "disks": [{
+            "imageName": "longhorn-qa/image-nbv7f",
```

Comment:
We might need a better image name like `ubuntu-22-04`. WDYT?

Reply:
It looks like it expects an image id instead of an image name?

```
docker-machine-driver-harvester: ELF 64-bit LSB executable, x86-64, version
    1 (SYSV), statically linked, stripped

    Trying to access option  which does not exist

    THIS ***WILL*** CAUSE UNEXPECTED BEHAVIOR

    Type assertion did not go smoothly to string for key 

    Running pre-create checks...

    Error with pre-create check: "virtualmachineimages.harvesterhci.io
    \"ubuntu-2204\" not found"

    The default lines below are for a sh/bash shell, you can specify the shell
    you're using, with the --shell flag.
  failureReason: CreateError
```